### PR TITLE
Clickhouse client and clickhouse local added queries-file parameter

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

https://github.com/ClickHouse/ClickHouse/issues/4491

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added `queries-file` parameter for clickhouse-client and clickhouse-local.

### CODE REVIEW

File: programs/client/Client.cpp [hunk 1]
```diff
@@ -469,7 +470,9 @@ class Client : public Poco::Util::Application
         ///   The value of the option is used as the text of query (or of multiple queries).
         ///   If stdin is not a terminal, INSERT data for the first query is read from it.
         /// - stdin is not a terminal. In this case queries are read from it.
-        if (!stdin_is_a_tty || config().has("query"))
+        /// - -qf (--queries-file) command line option is present.
+        ///   The value of the option is used as file with query (or of multiple queries) to execute.
+        if (!stdin_is_a_tty || config().has("query") || config().has("queries-file"))
```
Comment (kitaisreal) [inline programs/client/Client.cpp:475]: @alexey-milovidov should we handle case and throw error if user specify both 'query' and 'queries-file' ?

**Comment by akuzm** [comment]
By the way, does it already work with shell redirection, e.g. `clickhouse-client < queries.sql`?
Looks like this is the only way in MySQL, and Posgres supports both but they are "subtly different"

https://www.postgresql.org/docs/current/app-psql.html (see the --file option)
https://dev.mysql.com/doc/refman/8.0/en/mysql-batch-commands.html

File: programs/client/Client.cpp (hunk 1 above)
Comment (filimonov) [reply programs/client/Client.cpp:475]: And what is the other option? All alternatives i can imagine sound counter-intuitive

**Comment by kitaisreal** [comment]
> By the way, does it already work with shell redirection, e.g. `clickhouse-client < queries.sql`?
> Looks like this is the only way in MySQL, and Posgres supports both but they are "subtly different"
> 
> https://www.postgresql.org/docs/current/app-psql.html (see the --file option)
> https://dev.mysql.com/doc/refman/8.0/en/mysql-batch-commands.html

It will work with ./clickhouse-client --multiquery < 'file_name'

**Comment by kitaisreal** [comment]
> You need to create at least some basic tests for that.
> 
> Check that as an example:
> https://github.com/ClickHouse/ClickHouse/blob/c48da8d544fe271a0c2b16c7e96d583365442f1c/tests/queries/0_stateless/00366_multi_statements.sh
> https://github.com/ClickHouse/ClickHouse/blob/c48da8d544fe271a0c2b16c7e96d583365442f1c/tests/queries/0_stateless/00366_multi_statements.reference

Can I create temp query files in stateless tests ?

File: docs/en/interfaces/cli.md [hunk 2]
```diff
@@ -114,6 +114,7 @@ You can pass parameters to `clickhouse-client` (all parameters have a default va
 -   `--user, -u` – The username. Default value: default.
 -   `--password` – The password. Default value: empty string.
 -   `--query, -q` – The query to process when using non-interactive mode.
+-   `--queries-file, -qa` - file path with queries to execute.
```
[the remark this answers is not in the document]
Comment (kitaisreal) [reply docs/en/interfaces/cli.md]: Fixed. Thanks.

File: programs/local/LocalServer.cpp [hunk 3]
```diff
@@ -177,14 +179,20 @@ try
...
         if (config().hasOption("verbose"))
             std::cerr << "There are no queries to process." << '\n';
 
         return Application::EXIT_OK;
     }
 
+    if (config().has("query") && config().has("queries-file"))
+    {
+        std::cerr << "Specify either `query` or `queries-file` option" << std::endl;
+        return Application::EXIT_OK;
```
Comment (filimonov) [inline programs/local/LocalServer.cpp:193]: it's not OK :) 

```suggestion
        throw Exception("Specify either query either queries-file option", ErrorCodes::BAD_ARGUMENTS);
```

File: programs/client/Client.cpp [hunk 4]
```diff
@@ -779,8 +788,15 @@ class Client : public Poco::Util::Application
     {
         String text;
 
-        if (config().has("query"))
-            text = config().getRawString("query");  /// Poco configuration should not process substitutions in form of ${...} inside query.
+        if (config().has("queries-file"))
+        {
+            ReadBufferFromFile in(config().getRawString("queries-file"));
```
Comment (filimonov) [inline programs/client/Client.cpp:793]:
```suggestion
            ReadBufferFromFile in(config().getString("queries-file"));
```

File: programs/client/Client.cpp [hunk 5]
```diff
@@ -469,9 +470,17 @@ class Client : public Poco::Util::Application
...
-        if (!stdin_is_a_tty || config().has("query"))
+        /// - -qf (--queries-file) command line option is present.
+        ///   The value of the option is used as file with query (or of multiple queries) to execute.
+        if (!stdin_is_a_tty || config().has("query") || config().has("queries-file"))
             is_interactive = false;
 
+        if (config().has("query") && config().has("queries-file"))
+        {
+            std::cerr << "Specify either `query` or `queries-file` option" << std::endl;
+            return Application::EXIT_OK;
```
Comment (filimonov) [inline programs/client/Client.cpp:481]: it's not OK :) 

```suggestion
        throw Exception("Specify either query either queries-file option", ErrorCodes::BAD_ARGUMENTS);
```

File: programs/local/LocalServer.cpp [hunk 6]
```diff
@@ -321,7 +329,17 @@ std::string LocalServer::getInitialCreateTableQuery()
...
     String initial_create_query = getInitialCreateTableQuery();
-    String queries_str = initial_create_query + config().getRawString("query");
+    String queries_str = initial_create_query;
+
+    if (config().has("query"))
+        queries_str += config().getRawString("query");
+    else
+    {
+        String queries_from_file;
+        ReadBufferFromFile in(config().getRawString("queries-file"));
```
Comment (filimonov) [inline programs/local/LocalServer.cpp:339]:
```suggestion
        ReadBufferFromFile in(config().getString("queries-file"));
```

**Review by filimonov — CHANGES_REQUESTED** [review]

File: programs/local/LocalServer.cpp (hunk 6 above)
Comment (kitaisreal) [reply programs/local/LocalServer.cpp]: Fixed.

File: programs/client/Client.cpp (hunk 5 above)
Comment (kitaisreal) [reply programs/client/Client.cpp]: Fixed.

File: programs/client/Client.cpp (hunk 4 above)
Comment (kitaisreal) [reply programs/client/Client.cpp]: Fixed.

File: programs/local/LocalServer.cpp (hunk 3 above)
Comment (kitaisreal) [reply programs/local/LocalServer.cpp]: Fixed.

**Review by filimonov — APPROVED** [review]
Lgtm

**Comment by kitaisreal** [comment]
@alexey-milovidov did I break AST fuzzer or this is masters problem ?

**Pull request merged**